spec/hyacinth/adapters/preservation_adapter/fedora3/system_spec.rb

```diff
@@ -0,0 +1,172 @@
+require 'rails_helper'
+
+describe Hyacinth::Adapters::PreservationAdapter::Fedora3, fedora: true do
+  let(:object_pid) { "test:1" }
+  let(:location_uri) { "fedora3://" + object_pid }
+  let(:pid_generator) { instance_double(PidGenerator) }
+  let(:rubydora_config) { Rails.application.config_for(:fedora) }
+  let(:adapter_args) { rubydora_config.merge(pid_generator: pid_generator) }
+
+  let(:adapter) do
+    described_class.new(adapter_args)
+  end
+  let(:dc_xml_src) { file_fixture("fedora3/update-dc.xml").read }
+
+  before :all do
+    Rubydora.repository = Rubydora.connect(Rails.application.config_for(:fedora))
+  end
+
+  before do
+    # create the test object
+    Rubydora::DigitalObject.create(object_pid)
+  end
+  after do
+    # destroy the test object
+    Rubydora::DigitalObject.new(object_pid).delete
+  end
+
+  describe "#location_uri_to_fedora3_pid" do
+    subject { adapter.location_uri_to_fedora3_pid(location_uri) }
+    it { is_expected.to eql(object_pid) }
+  end
+
+  describe "#exists?" do
+    context "an object exists" do
+      it { expect(adapter.exists?(location_uri)).to be true }
+    end
+
+    context "an object does not exist" do
+      let(:location_uri) { "fedora3://" + object_pid + 'doesnotexist' }
+      it { expect(adapter.exists?(location_uri)).to be false }
+    end
+  end
+  describe "#generate_new_location_uri" do
+    subject { adapter.generate_new_location_uri }
+    before do
+      allow(pid_generator).to receive(:next_pid).and_return(*([1, 2].map { |x| "test:#{x}" }))
+    end
+    it "returns the first new location URI" do
+      is_expected.to eql("fedora3://test:2")
+    end
+  end
+  describe "#persist_impl" do
+    let(:rubydora_object) { Rubydora::DigitalObject.new(object_pid, Rubydora.repository) }
+    let(:hyacinth_object) { DigitalObject::Item.new }
+    context "core data" do
+      it "persists core data to Fedora3" do
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        actual_json = JSON.parse(rubydora_object.datastreams[described_class::HYACINTH_CORE_DATASTREAM_NAME].content)
+        expect(actual_json).to have_key("uid")
+      end
+    end
+    context "field exports" do
+      let(:digital_object_title) { "Assigned Label" }
+      before do
+        FactoryBot.create(:export_rule) # creates descMetadata
+      end
+      it "persists templated field exports to datastreams" do
+        hyacinth_object.dynamic_field_data['name'] = [{ 'role' => "Farmer" }]
+        adapter.persist_impl("fedora3://#{object_pid}", hyacinth_object)
+        actual_xml = rubydora_object.datastreams["descMetadata"].content
+        actual_xml.sub!(/^<\?.+\?>/, '') # remove PI
+        actual_xml.gsub!('mods:', '') # remove ns
+        actual_xml.gsub!(/\s/, '') # remove whitespace
+        expect(actual_xml).to eql("<mods><name>Farmer</name></mods>")
+      end
+      it "makes a core object property assignments" do
```

Comment:
Is this test meant to prove the successful assignment of one property or two (with state being implied as the second one, thus the `rubydora_object.state` equals `'A'` check)?  The test label threw me off, and I wasn't sure if you meant to include a `rubydora_object.state` check in this test.